Source/JavaScriptCore/dfg/DFGInPlaceAbstractState.cpp

```diff
@@ -370,14 +377,18 @@ bool InPlaceAbstractState::merge(BasicBlock* from, BasicBlock* to)
             unsigned valueCountInFromBlock = 0;
             for (NodeAbstractValuePair& fromBlockValueAtTail : from->ssa->valuesAtTail) {
                 if (fromBlockValueAtTail.node == node) {
-                    ASSERT(fromBlockValueAtTail.value == forNode(node));
+                    if (node->op() == EnumeratorNextUpdateIndexAndMode)
+                        ASSERT(hasClearedAbstractState(node) && !fromBlockValueAtTail.value);
+                    else
+                        ASSERT(fromBlockValueAtTail.value == forNode(node));
                     ++valueCountInFromBlock;
                 }
             }
             ASSERT(valueCountInFromBlock == 1);
 #endif
 
-            changed |= entry.value.merge(forNode(node));
+            if (node->op() != EnumeratorNextUpdateIndexAndMode)
+                changed |= entry.value.merge(forNode(node));
```

Comment:
Do we need this? (And if so, let's use `isTuple` instead).

Reply:
Yes we need that since `forNode` has `ASSERT(!node->isTuple());`.